client/my-sites/hosting/main.js

```diff
@@ -53,6 +54,6 @@ export default connect( state => {
 	const siteId = getSelectedSiteId( state );
 
 	return {
-		isDisabled: ! isSiteAutomatedTransfer( state, siteId ),
+		isDisabled: ! isSiteAutomatedTransfer( state, siteId ) || ! canSiteViewAtomicHosting( state ),
```

Comment:
Can we maybe do the `canSiteViewAtomicHosting` when rendering the component? That way, the `Hosting` component won't be rendered at all when switching to a ineligible site:

```js
const Hosting = ( { translate, isDisabled, canViewAtomicHosting } ) => {
	if ( ! canViewAtomicHosting ) {
		return null;
	}
	// ...
}

export default connect( state => {
	const siteId = getSelectedSiteId( state );

	return {
		isDisabled: ! isSiteAutomatedTransfer( state, siteId ),
		canViewAtomicHosting: canSiteViewAtomicHosting( state, siteId ),
	}
}
```

Reply:
Nice, that fixes it 👍